utils.py

```diff
@@ -0,0 +1,165 @@
+import asyncio
+import time
+
+import aiohttp
+
+from bs4 import BeautifulSoup
+
+from config import Config
+
+
+SNIPPET_FETCH_TIMEOUT_S = 3  # in case some webpage not respond
+
+google_search = 'https://customsearch.googleapis.com/customsearch/v1?key={}&cx={}&num=10'\
+    .format(Config.GOOGLE_API, Config.GOOGLE_CX)
+
+bing_search = 'https://api.bing.microsoft.com/v7.0/search/?responseFilter=Webpages&count=10'
+bing_header = {'Ocp-Apim-Subscription-Key': Config.BING_API_KEY}
+
+
+async def async_main_searches(requests):
+    async with aiohttp.ClientSession() as session:
+        tasks = [async_get_url_json(session, url=req['url'], header=req['header']) for req in requests]
+        return await asyncio.gather(*tasks)
+
+
+async def async_main_snippets(websites):
+    async with aiohttp.ClientSession() as session:
+        tasks = [async_get_custom_snippet(session, website, websites['query']) for website in websites['searchRes']]
+        return await asyncio.gather(*tasks)
+
+
+async def async_get_url_json(session, url, header):
+    ts = time.time()
+    async with session.get(url, headers=header) as resp:
+        if resp.status == 200:
+            resp_json = await resp.json()
+            resp_json['reqRespTs'] = time.time() - ts
+            return resp_json
+        return None
+
+
+async def async_get_custom_snippet(session, website, query):
+    ts = time.time()
+    try:
+        timeout = aiohttp.ClientTimeout(total=SNIPPET_FETCH_TIMEOUT_S)
+        async with session.get(website['url'], timeout=timeout) as resp:
+            if resp.status == 200:
+                resp_content = await resp.text(encoding='utf-8')  # 'windows-1251, resp.content.read()
+                website['customSnippetFetchTs'] = time.time() - ts
+
+                soup = BeautifulSoup(resp_content, 'html.parser')
+                meta_tag = soup.find('meta', attrs={'name': 'description'})
+                if meta_tag:
+                    website['customSnippet'] = meta_tag['content']
+                else:
+                    website['customSnippet'] = soup.title.string
+
+                # make query bold if found in custom snippet
+                try:
+                    query_start = website['customSnippet'].lower().index(query.lower())
+                    query_end = query_start + len(query) - 1
+                    html_snippet = website['customSnippet'][:query_start] + '<b>' + \
```

Comment:
You're just matching the whole query instead of using fuzzy search to match phrases/words possible in different grammatical forms. I didn't expect you to implement such a search, but there're plenty of libraries in Python doing that our of the box.